system-addon/test/unit/lib/Screenshots.test.js

```diff
@@ -49,6 +49,36 @@ describe("Screenshots", () => {
     });
   });
 
+  describe("#maybeCacheScreenshot", () => {
+    let link;
+    beforeEach(() => {
+      link = {__sharedCache: {updateLink: (prop, val) => { link[prop] = val; }}};
+    });
+    it("should call getScreenshotForURL", () => {
+      sandbox.stub(Screenshots, "getScreenshotForURL");
+      Screenshots.maybeCacheScreenshot(link, "mozilla.com", "image", sinon.stub());
+
+      assert.calledOnce(Screenshots.getScreenshotForURL);
+      assert.calledWithExactly(Screenshots.getScreenshotForURL, "mozilla.com");
+    });
+    it("should not call getScreenshotForURL twice if a fetch is in progress", () => {
+      sandbox.stub(Screenshots, "getScreenshotForURL").callsFake(() => new Promise(() => {}));
```

Comment:
My comment from before was `nit: .returns(new Promise(() => {}));`. Note the `.returns` instead of `.callsFake`. But this is fine…

Reply:
I missed that. Fixed now.